specification/search/data-plane/Azure.Search/preview/2024-09-01-preview/searchservice.json

```diff
@@ -0,0 +1,12622 @@
+      "enum": [
+        "cosine",
+        "euclidean",
+        "dotProduct",
+        "hamming"
+      ],
+      "x-ms-enum": {
+        "name": "VectorSearchAlgorithmMetric",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "cosine",
+            "name": "Cosine",
+            "description": "Measures the angle between vectors to quantify their similarity, disregarding magnitude. The smaller the angle, the closer the similarity."
+          },
+          {
+            "value": "euclidean",
+            "name": "Euclidean",
+            "description": "Computes the straight-line distance between vectors in a multi-dimensional space. The smaller the distance, the closer the similarity."
+          },
+          {
+            "value": "dotProduct",
+            "name": "DotProduct",
+            "description": "Calculates the sum of element-wise products to gauge alignment and magnitude similarity. The larger and more positive, the closer the similarity."
+          },
+          {
+            "value": "hamming",
+            "name": "Hamming",
+            "description": "Only applicable to bit-packed binary data types. Determines dissimilarity by counting differing positions in binary vectors. The fewer differences, the closer the similarity."
+          }
+        ]
+      },
+      "description": "The similarity metric to use for vector comparisons. It is recommended to choose the same similarity metric as the embedding model was trained on."
+    },
+    "VectorSearchAlgorithmKind": {
+      "type": "string",
+      "enum": [
+        "hnsw",
+        "exhaustiveKnn"
+      ],
+      "x-ms-enum": {
+        "name": "VectorSearchAlgorithmKind",
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "hnsw",
+            "name": "Hnsw",
+            "description": "HNSW (Hierarchical Navigable Small World), a type of approximate nearest neighbors algorithm."
+          },
+          {
+            "value": "exhaustiveKnn",
+            "name": "ExhaustiveKnn",
+            "description": "Exhaustive KNN algorithm which will perform brute-force search."
+          }
+        ]
+      },
+      "description": "The algorithm used for indexing and querying."
+    },
+    "VectorSearchCompressionConfiguration": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://learn.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name to associate with this particular configuration.",
+          "x-nullable": false
+        },
+        "kind": {
+          "$ref": "#/definitions/VectorSearchCompressionKind",
+          "description": "The name of the kind of compression method being configured for use with vector search.",
+          "x-nullable": false
+        },
+        "storeOriginalVectors": {
```

Comment:
We'll postpone "HNSW Storage Optimization" to Nov. release, so please help remove "storeOriginalVectors" from 2024-09-01-preview swagger.